controllers/provider-aws/pkg/awsclient/client.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+// NewClient creates a new Client for the given AWS credentials <accessKeyID>, <secretAccessKey>, and
+// the AWS region <region>.
+// It initializes the clients for the various services like EC2, ELB, etc.
+func NewClient(accessKeyID, secretAccessKey, region string) ClientInterface {
+	var (
+		awsConfig = &aws.Config{
+			Credentials: credentials.NewStaticCredentials(accessKeyID, secretAccessKey, ""),
+		}
+		sess   = session.Must(session.NewSession(awsConfig))
+		config = &aws.Config{Region: aws.String(region)}
+	)
+
+	return &Client{
+		EC2: ec2.New(sess, config),
+		ELB: elb.New(sess, config),
+		STS: sts.New(sess, config),
+	}
+}
+
+// GetAccountID returns the ID of the AWS account the Client is interacting with.
+func (c *Client) GetAccountID(ctx context.Context) (string, error) {
+	getCallerIdentityOutput, err := c.STS.GetCallerIdentityWithContext(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return "", err
+	}
+	return *getCallerIdentityOutput.Account, nil
+}
+
+// GetInternetGateway returns the ID of the internet gateway attached to the given VPC <vpcID>.
+// If there is no internet gateway attached, the returned string will be empty.
+func (c *Client) GetInternetGateway(ctx context.Context, vpcID string) (string, error) {
+	describeInternetGatewaysInput := &ec2.DescribeInternetGatewaysInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("attachment.vpc-id"),
+				Values: []*string{
+					aws.String(vpcID),
+				},
+			},
+		},
+	}
+	describeInternetGatewaysOutput, err := c.EC2.DescribeInternetGatewaysWithContext(ctx, describeInternetGatewaysInput)
+	if err != nil {
+		return "", err
+	}
+
+	if describeInternetGatewaysOutput.InternetGateways != nil {
+		if *describeInternetGatewaysOutput.InternetGateways[0].InternetGatewayId == "" {
+			return "", fmt.Errorf("no attached internet gateway found for vpc %s", vpcID)
+		}
+		return *describeInternetGatewaysOutput.InternetGateways[0].InternetGatewayId, nil
+	}
+	return "", fmt.Errorf("no attached internet gateway found for vpc %s", vpcID)
+}
+
+// The following functions are only temporary needed due to https://github.com/gardener/gardener/issues/129.
+
+// ListKubernetesELBs returns the list of load balancers in the given <vpcID> tagged with <clusterName>.
+func (c *Client) ListKubernetesELBs(ctx context.Context, vpcID, clusterName string) ([]string, error) {
+	output, err := c.ELB.DescribeLoadBalancersWithContext(ctx, &elb.DescribeLoadBalancersInput{})
+	if err != nil {
+		return nil, err
+	}
+
+	var results []string
+	for _, lb := range output.LoadBalancerDescriptions {
+		if lb.VPCId != nil && *lb.VPCId == vpcID {
+			tags, err := c.ELB.DescribeTagsWithContext(ctx, &elb.DescribeTagsInput{
```

Comment:
Can you write a `TODO` to optimize this later? 
`DescribeTagsWithContext` can take multiple LoadBalancers and you can probably make just 1 call to collect all Tags